### PR TITLE
Add DiagnosticLoggingChangeHandler.

### PR DESCRIPTION
**Pull request by jamesross3**
This change is [Reviewable](https://reviewable.io/reviews/palantir/witchcraft-go-server/197)

### CODE REVIEW

**Comment by jamesross3** [comment]
addresses https://github.com/palantir/witchcraft-go-server/issues/194
Right now, this will only emit a diagnostic log when health becomes worse than HealthStateRepairing. This feels like a reasonable starting place. There is nothing stopping us from adding a constructor param in the future that allows custom logic for when a diagnostic log is emitted.

**Review by bmoylan — APPROVED** [review]
> Reviewed 12 of 13 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/palantir/witchcraft-go-server/197)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by bmoylan** [comment]
@jamesross3 I guess changelog-app can't write to your fork

**Comment by jamesross3** [comment]
whoops thanks @bmoylan, added changelog

release once merge pls?

**Comment by bmoylan** [comment]
did you actually push the changelog?

File: changelog/@unreleased/pr-195.v2.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 type: feature
 feature:
   description: |
-    Add HealthStatusChangeHandler interface. Invoke health status change handlers whenever health checks have changed state.
+    Add DiagnosticLoggingChangeHandler.
   links:
-    - https://github.com/palantir/witchcraft-go-server/pull/195
+    - https://github.com/palantir/witchcraft-go-server/pull/197
```
Comment (bmoylan) [inline changelog/@unreleased/pr-195.v2.yml]: why are we changing this file instead of creating a new one?

**Review by bmoylan — APPROVED** [review]

**Pull request closed without merging**